libr/main/radare2.c

```diff
@@ -391,7 +391,17 @@ R_API int r_main_radare2(int argc, const char **argv) {
 	if (e) {
 		RStrBuf *sb = r_strbuf_new (NULL);
 		while (e && *e) {
-			r_strbuf_appendf (sb, "setenv=%s\n", *e);
+			char *k = strdup (*e);
+			char *v = strchr (k, '=');
+			if (v) {
+				*v++ = 0;
+				v = r_str_escape (v);
```

Comment:
`r_str_escape` strips anything that look like terminal color codes. 

Weird I thought I made this comment twice already but it wasn't here.

Reply:
Thanks again and sorry for sounding a bit rude previously as I did not think of the scenario of having newline as values in env vars. I will look into this again.

Reply:
I have tested color codes and escape sequence and it should work now.